src/app/dim-ui/Sheet.tsx

```diff
@@ -225,20 +183,62 @@ export default function Sheet({
         dragHandle.current?.contains(e.target as Node) ||
         sheetContents.current!.scrollTop === 0
       ) {
-        dragging.current = true;
+        dragControls.start(e);
       }
     },
-    []
+    [dragControls]
   );
 
-  const dragHandleUp = useCallback(() => (dragging.current = false), []);
+  // When drag ends we determine if the sheet should be closed either via the final
+  // drag velocity or if the sheet has been dragged halfway the down from its height.
+  const handleDragEnd = useCallback(
+    (_event: TouchEvent | MouseEvent | PointerEvent, info: PanInfo) => {
+      if (
+        info.velocity.y > dismissVelocity ||
+        (sheet.current && info.offset.y > dismissAmount * sheet.current.clientHeight)
+      ) {
+        triggerClose();
+        return;
+      }
+      animationControls.start('open');
+    },
+    [animationControls, triggerClose]
+  );
+
+  useLayoutEffect(() => {
+    if (freezeInitialHeight && sheetContents.current && !frozenHeight) {
+      if (sheetContents.current.clientHeight > 0) {
+        setFrozenHeight(sheetContents.current.clientHeight);
+      } else {
+        setTimeout(() => {
+          sheetContents.current && setFrozenHeight(sheetContents.current.clientHeight);
+        }, 500);
+      }
+    }
+  }, [freezeInitialHeight, frozenHeight]);
+
+  useEffect(() => {
+    animationControls.start('open');
+  }, [animationControls]);
 
   return (
     <SheetDisabledContext.Provider value={setParentDisabled}>
       <PressTipRoot.Provider value={sheet}>
-        <animated.div
-          {...bindDrag()}
-          style={{ ...springProps, touchAction: 'none', zIndex }}
```

Comment:
I don't think we need the `touch-action: none` here anymore. It is now on the drag handle element though as per my comment regarding motion needing it